### PR TITLE
Replace parsec with megaparsec.

### LINKED ISSUE #39: Transition to Megaparsec
Megaparsec provides for much needed better error messages.

### PR DESCRIPTION
**Pull request by athas**
The main advantage is slightly better error messages.

Closes #39.

### CODE REVIEW

**Review by oleks — APPROVED** [review]
That was way easier than I expected!

File: src/Parser/ImplParsec.hs [hunk 1]
```diff
@@ -124,7 +119,7 @@ parseFeedback depth = do
 
 parseRegularJudgement :: Int -> String -> MrkParser Judgement
 parseRegularJudgement depth title = do
-  void $ spaces
+  void $ space
```
Comment (oleks) [inline src/Parser/ImplParsec.hs]: I guess void is unnecessary here?

**Comment by oleks** [comment]
Thanks for purging the ReadP parser as well!

File: src/Parser/ImplParsec.hs (hunk 1 above)
Comment (athas) [reply src/Parser/ImplParsec.hs]: Yes, most of them are now unnecessary.

**Pull request merged**